### PR TITLE
BUG: GeoSeries constructor should return Series for empty array with dtype when possible (fix GeoSeries.isna for empty Series)

### PR DESCRIPTION
**Pull request by jorisvandenbossche**
While looking into some more of the warnings from our tests, I noticed that `GeoSeries([]).isna()` didn't properly return a boolean Series:

```
>>> geopandas.GeoSeries([]).isna()
GeoSeries([], dtype: geometry)
```

while the result of `isna` should always be boolean.

### CODE REVIEW

File: geopandas/tests/test_geoseries.py [hunk 1]
```diff
@@ -434,6 +441,35 @@ def test_data_is_none(self):
...
+            # this gets converted to object dtype by pandas
+            # np.array([], dtype="str"),
+        ]:
+            with pytest.warns(FutureWarning):
+                s = GeoSeries(arr)
+            assert not isinstance(s, GeoSeries)
+            assert type(s) == pd.Series
+
+        # dtypes that can potentially hold geometry-like data (object) or
+        # can come from emtpy data (float64)
```
Comment (martinfleis) [inline geopandas/tests/test_geoseries.py:463]:
```suggestion
        # can come from empty data (float64)
```

File: geopandas/tests/test_geoseries.py [hunk 2]
```diff
@@ -434,6 +441,35 @@ def test_data_is_none(self):
...
+        for arr in [
+            np.array([], dtype="object"),
+            np.array([], dtype="float64"),
+            np.array([], dtype="str"),
+        ]:
+            with pytest.warns(None):
+                s = GeoSeries(arr)
```
Comment (martinfleis) [inline geopandas/tests/test_geoseries.py:470]: I _think_ that this does nothing. If you want to check that no warning is emitted here, you need

```suggestion
            with pytest.warns(None) as record:
                s = GeoSeries(arr)
            assert not record
```

**Review by martinfleis — COMMENTED** [review]
Thanks! Two minor suggestions in tests.

File: geopandas/tests/test_geoseries.py (hunk 2 above)
Comment (jorisvandenbossche) [reply geopandas/tests/test_geoseries.py]: Ah, yes, thanks!

**Pull request merged**